### PR TITLE
Update pending-documentation

### PR DESCRIPTION
**Pull request by radumereuta**
with more LLVM debug info

### CODE REVIEW

File: pending-documentation.md [hunk 1]
```diff
@@ -1361,6 +1367,15 @@ evaluating another function, we would also be able to see the arguments of that
 function as well, unless the function was tail recursive, in which case no
 stack frame would exist once the tail call was performed.
 
+### Breaking on a set of rules or functions
```
Comment (ehildenb) [inline pending-documentation.md:1365]:
```suggestion
### Breaking on a set of rules or functions

```

File: pending-documentation.md [hunk 2]
```diff
@@ -1361,6 +1367,15 @@ evaluating another function, we would also be able to see the arguments of that
...
+Using `rbreak <regex>` you can set breakpoints on multiple functions.
+
+`rbreak Lbl` - sets a breakpoint on all non hooked `function`s
```
Comment (ehildenb) [inline pending-documentation.md]:
```suggestion
-   `rbreak Lbl` - sets a breakpoint on all non hooked `function`s
```

File: pending-documentation.md [hunk 3]
```diff
@@ -1361,6 +1367,15 @@ evaluating another function, we would also be able to see the arguments of that
...
+
+`rbreak Lbl.*TEST` - sets a breakpoint on all `function`s from module `TEST`
```
Comment (ehildenb) [inline pending-documentation.md]:
```suggestion
-   `rbreak Lbl.*TEST` - sets a breakpoint on all `function`s from module `TEST`
```

File: pending-documentation.md [hunk 4]
```diff
@@ -1361,6 +1367,15 @@ evaluating another function, we would also be able to see the arguments of that
...
+
+`rbreak hook_INT` - sets a breakpoint on all hooks from module `INT`
```
Comment (ehildenb) [inline pending-documentation.md]:
```suggestion
-   `rbreak hook_INT` - sets a breakpoint on all hooks from module `INT`
```

File: pending-documentation.md (hunk 1 above)
Comment (radumereuta) [reply pending-documentation.md:1365]: What's the difference here?

Comment (ehildenb) [reply pending-documentation.md:1365]: Adding a newline after the level-3 header.

File: pending-documentation.md [hunk 5]
```diff
@@ -1179,6 +1179,10 @@ on a particular program by passing the `--debugger` flag to krun, or by
 invoking the llvm backend interpreter directly. Below we provide a simple
 tutorial to explain some of the basic commands supported by the LLVM backend.
 
+Before that make sure to build K with the option `-Dproject.build.type=Debug`.
+Or modify `<kdir>/llvm-backend/pom.xml` and make sure `<project.build.type>` 
+is set to `Debug`.
```
Comment (nishantjr) [inline pending-documentation.md]: This shouldn't be needed once the default is `RelWithDebInfo`

Comment (radumereuta) [reply pending-documentation.md]: That PR was closed without being merged.

File: pending-documentation.md [hunk 6]
```diff
@@ -1361,6 +1362,27 @@ evaluating another function, we would also be able to see the arguments of that
...
+
+-   `rbreak Lbl` - sets a breakpoint on all non hooked `function`s
+
+-   `rbreak Lbl.*TEST` - sets a breakpoint on all `function`s from module `TEST`
+
+-   `rbreak hook_INT` - sets a breakpoint on all hooks from module `INT`
+
+### Other debugger issues
+
+-   `Python Exception <class 'gdb.error'> No type named string`
```
Comment (dwightguth) [inline pending-documentation.md]: this line can be removed as we now always build the backend with debug symbols in the runtime, even in release configuration.

**Review by dwightguth — CHANGES_REQUESTED** [review]

File: pending-documentation.md (hunk 6 above)
Comment (radumereuta) [reply pending-documentation.md]: It doesn't work though. Latest version from master:
```
radu@ubuntu:~/work/test$ rm -rf test-kompiled/
radu@ubuntu:~/work/test$ kompile test.k -ccopt -g -ccopt -O1
radu@ubuntu:~/work/test$ krun --output kast --debugger
(gdb) break definition.kore:step
Breakpoint 1 at 0x215310
(gdb) r
Breakpoint 1, 0x0000000000215310 in step (Python Exception <class 'gdb.error'> No type named string.: 
subject=0x155552e001a0)
```
Changed to Debug in llvm-backend/pom.xml
```
radu@ubuntu:~/work/test$ rm -rf test-kompiled/
radu@ubuntu:~/work/test$ kompile test.k -ccopt -g -ccopt -O1
radu@ubuntu:~/work/test$ krun --output kast --debugger
(gdb) break definition.kore:step
Breakpoint 1 at 0x28c2b0
(gdb) r
Breakpoint 1, 0x000000000028c2b0 in step (subject=
    `<generatedTop>{}`(`<k>{}`(`kseq{}`(Lblinit{}(.KList),`kseq{}`(Lblsmt{}(.KList),dotk{}(.KList)))),`<generatedCounter>{}`(#token("0", "Int"))))
```

**Review by dwightguth — APPROVED** [review]

**Pull request merged**